### ISSUE TITLE
If i pull update i get "Skipped - Image is already present locally"

### ISSUE THREAD

**Issue by paolo1983** [docker/compose#9791]
Ho to all. I got upgrade to last docker-compose (2.9.0.1) and now i'm not able to upgrade containers. When i do "docker-compose pull xxx" i get "Skipped - Image is already present locally". I had to downgrade to 2.6.1.1 to upgrade my container. Is It a bug or with the new version i have to add a parameters? For example i want to upgrade home assistant and in the yaml the image Is tagged with :stable

**Comment by laurazard** [comment]
Hi @paolo1983, thanks for the report.

This was a known issue in `v2.9`, and has been fixed in `v2.10.1`. `compose pull` should now always try to pull the latest version.

**Comment by paolo1983** [comment]
I'm on 2.10.1 but i have the same issue, can you check?

**Comment by squidds** [comment]
It seems to be fixed in 2.10.2

> Linux
> You can download Docker Compose binaries from the [release page](https://github.com/docker/compose/releases) on this repository.
> 
> Rename the relevant binary for your OS to docker-compose and copy it to $HOME/.docker/cli-plugins
> 
> Or copy it into one of these folders for installing it system-wide:
> 
> /usr/local/lib/docker/cli-plugins OR /usr/local/libexec/docker/cli-plugins
> /usr/lib/docker/cli-plugins OR /usr/libexec/docker/cli-plugins
> (might require to make the downloaded file executable with chmod +x)

From the compose V2 github. [https://github.com/docker/compose](https://github.com/docker/compose)

You can drop the 2.10.2 binary from [https://github.com/docker/compose/releases](https://github.com/docker/compose/releases) into the directory with your configuration file, make it executable, and then run. And it will pull new. If you cannot wait for your distribution to update to 2.10.2.

**Comment by laurazard** [comment]
Yes @squidds, the issue was introduced in `v2.10.0` and fixed `v2.10.1`, which is why I mentioned that version :)

**Comment by squidds** [comment]
OP's comment below your's says they were using version 2.10.1. Your comment said known in 2.9 but fixed in 2.10.1. I was simply showing that if he absolutely had to, he could pull with the binary.

**Comment by laurazard** [comment]
Totally! I was also wrong when I said introduced in `v2.9` in the first comment, and then I didn't notice it 😅. It was introduced in `v2.10.0` in https://github.com/docker/compose/pull/9173 and fixed in `v2.10.1` in https://github.com/docker/compose/issues/9773

**Comment by paolo1983** [comment]
Now it works! Thanks
